packages/code-du-travail-frontend/src/outils/indemniteLicenciement/SalaireTempsPlein.js

```diff
@@ -0,0 +1,132 @@
+import React from "react";
+import PropTypes from "prop-types";
+
+import { Input } from "./stepStyles";
+
+const times = num => Array.from({ length: num }, (_, i) => i);
+
+class SalaireTempsPlein extends React.Component {
+  static propTypes = {
+    onChange: PropTypes.func,
+    value: PropTypes.shape({
+      isPartiel: PropTypes.bool,
+      periods: PropTypes.array,
+      derniersMois: PropTypes.arrayOf(PropTypes.number)
+    }).isRequired
+  };
+  state = {
+    salaires: this.props.value.derniersMois
+  };
+
+  onSalaireClick = i => e => {
+    const value = parseFloat(e.target.value);
+    if (value === 0) {
+      this.updateSalaireValue(i, "");
+    }
+  };
+
+  onSalaireBlur = i => e => {
+    // update below fields values when empty
+    const value = parseFloat(e.target.value);
+    if (!value) {
+      this.updateSalaireValue(i, 0);
+      return;
+    }
+    const curSalaires = [...this.state.salaires];
+    let j = i + 1;
+    while (j < curSalaires.length) {
+      if (parseFloat(curSalaires[j]) > 0) {
+        break;
+      }
+      curSalaires[j] = value;
+      j++;
+    }
```

Comment:
It looks like a 

```js
const firstSalaireIndex = this.state.salaires
  .slice(index + 1)
  .map(parseFloat)
  .findIndex(salaire => salaire > 0)
this.setState({salaire: this.state.salaires.slice(startIndex + 1, startIndex + firstSalaireIndex)
``` 

Is it ?

Reply:
I didn't  touch too much this since it was in the initial prototype but yeah we definitively could improve it

Reply:
I'm not sure it work

you can try it here https://codesandbox.io/s/lxjl3lxzl9

Reply:
```
thanks to  TDD, i ended up with return 
```js
this.state.salaires.map(
    (item, i) => (i >= index && parseFloat(item) === 0 ? value : item)
  );
```

Reply:
Looks better :ok_hand: 
Still I have no idea what it's for :laughing: